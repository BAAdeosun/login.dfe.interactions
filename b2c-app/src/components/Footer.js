import React from 'react'

export default function footer() {
    return (
        <div>
            I AM THE FOOTER
        </div>
    )
}
