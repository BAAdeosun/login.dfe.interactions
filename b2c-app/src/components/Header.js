import React from 'react'

function Header() {
    return (
        <header class="govuk-header " role="banner" data-module="govuk-header">
            <div class="govuk-header__container govuk-width-container">
                <div class="govuk-header__content">
                <a href="#" class="govuk-header__link govuk-header__link--service-name">
                    National Careers Service
                </a>
                <button type="button" class="govuk-header__menu-button govuk-js-header-toggle" aria-controls="navigation" aria-label="Show or hide Top Level Navigation">Menu</button>
                <nav>
                    <ul id="navigation" class="govuk-header__navigation " aria-label="Top Level Navigation">
                    <li class="govuk-header__navigation-item govuk-header__navigation-item--active">
                        <a class="govuk-header__link" href="#1">
                        Job profiles
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#2">
                        Skills health check
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#3">
                        Find a course
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#4">
                        Contact an adviser
                        </a>
                    </li>
                    <li class="govuk-header__navigation-item">
                        <a class="govuk-header__link" href="#4">
                        About us
                        </a>
                    </li>
                    </ul>
                </nav>
                </div>
            </div>
        </header>
    )
}

export default Header;