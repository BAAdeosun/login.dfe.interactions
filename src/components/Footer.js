import Link from 'next/link'

const Footer = () => (
    <footer>
        This is the footer
    </footer>
)

export default Footer