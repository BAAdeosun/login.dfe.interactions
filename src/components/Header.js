import Link from 'next/link'

const Header = () => (
    <header>
        This is the header
    </header>
)

export default Header