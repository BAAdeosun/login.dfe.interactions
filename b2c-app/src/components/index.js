import Header from './Header/Header';
import Footer from './Footer/Footer';
import B2C from './B2C/B2C';
import HeaderTitle from './HeaderTitle/HeaderTitle';
import PageNavigation from './PageNavigation/PageNavigation';
import NavigationLink from './NavigationLink/NavigationLink';
import Breadcrumbs from './Breadcrumbs/Breadcrumbs';
import MainContainer from './MainContainer/MainContainer';
import PageTitle from './PageTitle/PageTitle';
import FooterNavigation from './FooterNavigation/FooterNavigation';
import FooterLink from './FooterLink/FooterLink';
import PasswordHelpContainer from './PasswordHelp/PasswordHelpContainer';
import PasswordHelp from './PasswordHelp/PasswordHelp';
import PageLevelErrorContainer from './PageLevelErrorContainer/PageLevelErrorContainer';
import CreateNewPassword from './CreateNewPassword/CreateNewPassword';
import DateOfBirth from './DateOfBirth/DateOfBirth';
import TermsAndConditions from './TermsAndConditions/TermsAndConditions';
import ButtonLink from './ButtonLink/ButtonLink';
import Link from './Link/Link';
import Paragraph from './Paragraph/Paragraph';
import Postcode from './Postcode/Postcode';
import InputField from './InputField/InputField';

export default {
    Header: Header,
    Footer: Footer,
    B2C: B2C,
    HeaderTitle: HeaderTitle,
    PageNavigation: PageNavigation,
    NavigationLink: NavigationLink,
    Breadcrumbs: Breadcrumbs,
    MainContainer: MainContainer,
    PageTitle: PageTitle,
    FooterNavigation: FooterNavigation,
    FooterLink: FooterLink,
    PasswordHelpContainer: PasswordHelpContainer,
    PasswordHelp: PasswordHelp,
    PageLevelErrorContainer: PageLevelErrorContainer,
    CreateNewPassword: CreateNewPassword,
    DateOfBirth: DateOfBirth,
    TermsAndConditions: TermsAndConditions,
    ButtonLink: ButtonLink,
    Link: Link,
    Paragraph: Paragraph,
    Postcode: Postcode,
    InputField: InputField
}