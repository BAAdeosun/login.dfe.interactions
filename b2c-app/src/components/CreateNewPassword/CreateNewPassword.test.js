import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import components from '..';

it('renders without crashing', () => {
    shallow(<components.CreateNewPassword />);
});

it('renders correctly without props passed in', () => {
    const tree = renderer
        .create(<components.CreateNewPassword />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly with errors', () => {
    const tree = renderer.create(<components.CreateNewPassword showErrors={true} />);
    tree.root.instance.setState({
        errors: {
            newPassword: {
                visible: {
                    text: 'this is a test'
                }
            },
            reenteredPassword: {
                visible: {
                    text: 'this is a test'
                }
            }
        }
    });
    expect(tree.toJSON()).toMatchSnapshot()
});

it('calls validation, sets errors and calls onChange callback', () => {

    const mockOnChangeCallback = jest.fn();

    const wrapper = mount(<components.CreateNewPassword showErrors={true} onChange={mockOnChangeCallback} />);
    const newPasswordInput = wrapper.find('#newPasswordCustom');
    const reenteredPasswordInput = wrapper.find('#reenteredPasswordCustom');
    const validationSpy = jest.spyOn(wrapper.instance(), 'isValidPassword');

    const validPassword = 'aaaaaa1!';
    const invalidPassword = 'aaaaaaaaa';
    const shortPassword = 'a';
    let newPasswordChangeEvent = {
        target:
        {
            name: 'newPassword',
            value: shortPassword
        }
    };
    let reenteredPasswordChangeEvent = {
        target:
        {
            name: 'reenteredPassword',
            value: validPassword
        }
    };

    //default values set
    expect(wrapper.state().newPassword).toBe(null);
    expect(wrapper.state().reenteredPassword).toBe(null);
    expect(wrapper.state().errors.newPassword.current.text).toEqual('Enter your password');
    expect(wrapper.state().errors.reenteredPassword.current.text).toEqual('');

    //simulate entering a password too short
    newPasswordInput.simulate('change', newPasswordChangeEvent);
    //validation called
    expect(validationSpy).toHaveBeenCalled();
    //state value has been updated
    expect(wrapper.state().newPassword).toEqual(shortPassword);
    //check error message is set because reentered password is still not set
    expect(wrapper.state().errors.newPassword.current.text).toEqual('Enter a password between 8 and 16 characters');
    expect(wrapper.state().errors.reenteredPassword.current.text).toEqual('');
    //onChange callback function passed in has been called with the valid value
    expect(mockOnChangeCallback).toHaveBeenCalledWith({ password: null });

    //simulate entering a password that doesn't meet requirements
    newPasswordChangeEvent.target.value = invalidPassword;
    newPasswordInput.simulate('change', newPasswordChangeEvent);
    //validation called
    expect(validationSpy).toHaveBeenCalled();
    //state value has been updated
    expect(wrapper.state().newPassword).toEqual(invalidPassword);
    //check error message is set because reentered password is still not set
    expect(wrapper.state().errors.newPassword.current.text).toEqual('Invalid password');
    expect(wrapper.state().errors.reenteredPassword.current.text).toEqual('');
    //onChange callback function passed in has been called with the valid value
    expect(mockOnChangeCallback).toHaveBeenCalledWith({ password: null });

    //simulate entering valid password (but not reentered password yet)
    newPasswordChangeEvent.target.value = validPassword;
    newPasswordInput.simulate('change', newPasswordChangeEvent);
    //validation called
    expect(validationSpy).toHaveBeenCalled();
    //state value has been updated
    expect(wrapper.state().newPassword).toEqual(validPassword);
    //check error message is set because reentered password is still not set
    expect(wrapper.state().errors.newPassword.current.text).toEqual('');
    expect(wrapper.state().errors.reenteredPassword.current.text).toEqual('Re-enter your password');
    //onChange callback function passed in has been called with the valid value
    expect(mockOnChangeCallback).toHaveBeenCalledWith({ password: null });

    //simulate entering reentered password
    reenteredPasswordInput.simulate('change', reenteredPasswordChangeEvent);
    //validation called
    expect(validationSpy).toHaveBeenCalled();
    //state value has been updated
    expect(wrapper.state().reenteredPassword).toEqual(validPassword);
    //check error messages are cleared
    expect(wrapper.state().errors.newPassword.current.text).toEqual('');
    expect(wrapper.state().errors.reenteredPassword.current.text).toEqual('');
    //onChange callback function passed in has been called with the valid value
    expect(mockOnChangeCallback).toHaveBeenCalledWith({ password: validPassword });

    //simulate entering different password values
    reenteredPasswordChangeEvent.target.value = 'something';
    reenteredPasswordInput.simulate('change', reenteredPasswordChangeEvent);
    //validation called
    expect(validationSpy).toHaveBeenCalled();
    //state value has been updated
    expect(wrapper.state().reenteredPassword).toEqual('something');
    //check error message is set
    expect(wrapper.state().errors.newPassword.current.text).toEqual('');
    expect(wrapper.state().errors.reenteredPassword.current.text).toEqual('Your passwords do not match');
    //onChange callback function passed in has been called with null
    expect(mockOnChangeCallback).toHaveBeenCalledWith({ password: null });

});