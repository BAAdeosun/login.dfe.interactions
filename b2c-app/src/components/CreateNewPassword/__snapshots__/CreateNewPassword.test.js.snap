// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with errors 1`] = `
<div>
  <div
    className="govuk-form-group govuk-form-group--error"
  >
    <label
      className="govuk-label"
      htmlFor="newPasswordCustom"
    >
      Create new password
    </label>
    <span
      className="govuk-error-message"
      id="newPasswordError"
    >
      <span
        className="govuk-visually-hidden"
      >
        Error:
      </span>
      this is a test
    </span>
    <input
      className="govuk-input govuk-!-width-one-half"
      id="newPasswordCustom"
      name="newPassword"
      noValidate={true}
      onChange={[Function]}
      type="password"
    />
  </div>
  <div
    className="govuk-form-group"
  >
    <details
      className="govuk-details"
    >
      <summary
        className="govuk-details__summary"
      >
        <span
          className="govuk-details__summary-text"
        >
          Help choosing a valid password
        </span>
      </summary>
      <div
        className="govuk-details__text"
      >
        <div>
          <p
            className="govuk-body"
          >
            Your password must:
          </p>
          <ul
            className="govuk-list govuk-list--bullet"
          >
            <li>
              be between 8 and 16 characters
            </li>
            <li>
              include at least one uppercase letter and one lowercase letter
            </li>
            <li>
              include at least one number or one symbol
            </li>
          </ul>
        </div>
      </div>
    </details>
  </div>
  <div
    className="govuk-form-group govuk-form-group--error"
  >
    <label
      className="govuk-label"
      htmlFor="reenteredPasswordCustom"
    >
      Re-type password
    </label>
    <span
      className="govuk-error-message"
      id="reenteredPasswordError"
    >
      <span
        className="govuk-visually-hidden"
      >
        Error:
      </span>
      this is a test
    </span>
    <input
      className="govuk-input govuk-!-width-one-half"
      id="reenteredPasswordCustom"
      name="reenteredPassword"
      noValidate={true}
      onChange={[Function]}
      type="password"
    />
  </div>
</div>
`;

exports[`renders correctly without props passed in 1`] = `
<div>
  <div
    className="govuk-form-group "
  >
    <label
      className="govuk-label"
      htmlFor="newPasswordCustom"
    >
      Create new password
    </label>
    <input
      className="govuk-input govuk-!-width-one-half"
      id="newPasswordCustom"
      name="newPassword"
      noValidate={true}
      onChange={[Function]}
      type="password"
    />
  </div>
  <div
    className="govuk-form-group"
  >
    <details
      className="govuk-details"
    >
      <summary
        className="govuk-details__summary"
      >
        <span
          className="govuk-details__summary-text"
        >
          Help choosing a valid password
        </span>
      </summary>
      <div
        className="govuk-details__text"
      >
        <div>
          <p
            className="govuk-body"
          >
            Your password must:
          </p>
          <ul
            className="govuk-list govuk-list--bullet"
          >
            <li>
              be between 8 and 16 characters
            </li>
            <li>
              include at least one uppercase letter and one lowercase letter
            </li>
            <li>
              include at least one number or one symbol
            </li>
          </ul>
        </div>
      </div>
    </details>
  </div>
  <div
    className="govuk-form-group "
  >
    <label
      className="govuk-label"
      htmlFor="reenteredPasswordCustom"
    >
      Re-type password
    </label>
    <input
      className="govuk-input govuk-!-width-one-half"
      id="reenteredPasswordCustom"
      name="reenteredPassword"
      noValidate={true}
      onChange={[Function]}
      type="password"
    />
  </div>
</div>
`;
